compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -740,16 +752,27 @@ def reload_tasks(
                             fut.set_exception(funcx_err)
 
             if pending:
-                self._result_watcher = _ResultWatcher(self, port=self.amqp_port)
-                self._result_watcher.watch_for_task_results(pending)
-                self._result_watcher.start()
+                if rw is None:
+                    rw = self._get_result_watcher()
+                rw.watch_for_task_results(self, pending)
         else:
             log.warning(f"Received no tasks for Task Group ID: {task_group_id}")
 
-        # step 5: the goods for the consumer
+        # step 3: the goods for the consumer
         return futures
 
     def shutdown(self, wait=True, *, cancel_futures=False):
+        """Clean-up the resources associated with the Executor.
+
+        It is safe to call this method several times. Otherwise, no other methods
+        can be called after this one.
+
+        :param wait: If True, then this method will not return until all pending
+            futures have received results.
+        :param cancel_futures: If True, then this method will cancel all futures
```

Comment:
It's easy to conflate future with tasks. Adding a clarification that we are cancelling `futures` and we have no ability to cancel launched tasks might be helpful.

Reply:
Added some clarifying language.